### ISSUE TITLE
feat: In the `EstimatorReport`, specify which metric is actually used to do the `fit`

### ISSUE THREAD

**Issue by sylvaincom** [probabl-ai/skore#1217]
### Is your feature request related to a problem? Please describe.

The EstimatorReport currently (v0.6.1) displays a nice helper with suggestions of metrics. However, it is unclear which metric was actually used under the hood to fit the model.

<img src="https://github.com/user-attachments/assets/133f23ca-8105-4e69-9f84-046068917607" width="600">

### Describe the solution you'd like

Make it explicit which metric was used for the fit.

In the helper, maybe display something like `.accuracy(...) (★)` (but emojis will create problems as in https://github.com/probabl-ai/skore/issues/1076) or `.accuracy(...) (used for the fit)`. I don't really know.

### Additional context

Related to https://github.com/probabl-ai/skore/issues/578

**Comment by glemaitre** [comment]
It is not possible because the loss minimized might not simply be part of the metric used. Whenever you have a regularization or a complex model like ensemble, then there is no correspondence. Until scikit-learn does not expose the objective function, then I don't see it possible.

We spoke about exposing the objective function because it could be a potential feedback to show the evolution of the objective function over time during fit. In this case, we could offer to compute this objective function (but it is only meaningful on the training set and it would already be part of an estimator's attribute I think).
